### ISSUE TITLE
[Feature Request]: When generating the database helper code using the Connection, if a model has the $table property, then use that to create the helper code for the model

### ISSUE THREAD

**Issue by abishekrsrikaanth** [laravel-idea/plugin#822]
### Feature Description

When generating the database helper code using the Connection, if a model has the $table property, then use that to create the helper code for the model. Currently it tries to generate the table name from the Model name. Sometimes the model name may not be helpful to get the table name as the table name may be slightly different.

<img width="373" alt="CleanShot 2023-09-06 at 03 15 22@2x" src="https://github.com/laravel-idea/plugin/assets/1639302/2407e6e9-d911-4d5a-b90a-4bcecdd4afa2">

**Comment by adelf** [comment]
Could you show how you set the $table field?

**Comment by abishekrsrikaanth** [comment]
<img width="546" alt="CleanShot 2023-09-06 at 03 20 21@2x" src="https://github.com/laravel-idea/plugin/assets/1639302/8e6eb42b-b878-432c-bc80-80e300294693">

if the `$table` property doesn't exist, then you can try to generate it using the classname. I keep getting an error that some of the models couldn't generate because the corresponding table doesn't exist..

**Comment by adelf** [comment]
Laravel Idea will use $table value in this case. Any issues with it?

**Comment by abishekrsrikaanth** [comment]
I keep getting an error that some of the models couldn't generate because the corresponding table doesn't exist..

**Comment by adelf** [comment]
But you didn't set up any connection. The red error message is about it.

**Comment by abishekrsrikaanth** [comment]
No, that was just an example screenshot on a new laravel project that I haven't set up the connection yet.

**Comment by abishekrsrikaanth** [comment]
when corresponding migrations are there, they generate perfectly. but when I switch to **connection** mode and the class names don't correspond to the table name defined in `$table` property, then I get this error.

**Comment by abishekrsrikaanth** [comment]
Here is an example of the issue.
![CleanShot 2023-09-06 at 03 33 03@2x](https://github.com/laravel-idea/plugin/assets/1639302/db48bec9-2f0a-4acc-80bb-90208cf1644d)

and here is the model

<img width="354" alt="CleanShot 2023-09-06 at 03 34 56@2x" src="https://github.com/laravel-idea/plugin/assets/1639302/7d0cab65-1fc6-4663-8583-fa03b75dd7e2">

**Comment by adelf** [comment]
Does Cmd(Ctrl)-click on this field work?

Do you have this checkbox on?
<img width="698" alt="image" src="https://github.com/laravel-idea/plugin/assets/2818394/49ca4c39-7832-4faa-84c4-c401af86ac74">

**Comment by abishekrsrikaanth** [comment]
<img width="542" alt="CleanShot 2023-09-06 at 03 40 10@2x" src="https://github.com/laravel-idea/plugin/assets/1639302/148c698c-1ab7-4360-82b8-23e6dc32b63a">

**Comment by adelf** [comment]
Does Cmd(Ctrl)-click on this field work?

**Comment by robsontenorio** [comment]
> ### Feature Description
> When generating the database helper code using the Connection, if a model has the $table property, then use that to create the helper code for the model. Currently it tries to generate the table name from the Model name. Sometimes the model name may not be helpful to get the table name as the table name may be slightly different.
> 
> <img alt="CleanShot 2023-09-06 at 03 15 22@2x" width="373" src="https://user-images.githubusercontent.com/1639302/265833392-2407e6e9-d911-4d5a-b90a-4bcecdd4afa2.png">

Where can I find this settings ?

**Comment by mvdgeijn** [comment]
Same question here as @robsontenorio: Where can I find these settings in phpStorm (Windows)?

**Comment by sfinktah** [comment]
Erm, I think it is actually reading Models (and their $table property) now.  Now that I just found that you can use "Connections" rather than "Migrations" to select multiple datasources, I finally saw all 300+ tables load).

TIL: https://github.com/laravel-idea/plugin/issues/610#issuecomment-1218124991 )
